[ONNX] Add aten::zero support

titaiwangms opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #91731

Fixes #90268 

When we use `tensor.zero_()` with inplace slice, it actually uses `aten::zero` instead.

titaiwangms committed:
Add aten::zero support

[ghstack-poisoned]

BowenBao reviewed (approved):
🚀 nit: "[ONNX]" in title

titaiwangms commented:
@pytorchbot rebase

titaiwangms committed:
Update on "[ONNX] Add aten::zero support"


Fixes #90268 

When we use `tensor.zero_()` with inplace slice, it actually uses `aten::zero` instead. 

[ghstack-poisoned]

clee2000 commented:
@pytorchbot revert -m "failing test_correct_module_names https://hud.pytorch.org/pytorch/pytorch/commit/ff23508c0d491553dc8eea85fb45f49de52ca41f https://github.com/pytorch/pytorch/actions/runs/3859079162/jobs/6578419644" -c "ignoredsignal"

clee2000 commented:
@pytorchbot revert -m "failing test_correct_module_names https://hud.pytorch.org/pytorch/pytorch/commit/ff23508c0d491553dc8eea85fb45f49de52ca41f https://github.com/pytorch/pytorch/actions/runs/3859079162/jobs/6578419644" -c "ignoredsignal"

BowenBao commented:
```
# torch.onnx.symbolic_opset9.zero:
  - Is NOT public: it is not inside the module's (`torch.onnx.symbolic_opset9`) `__all__`
  - Does look public: it does look public because it follows the rules from the doc above (does not start with `_` and has a proper `__module__`).
  - You can do either of these two things to fix this problem:
    - To make it public: add it from the modules's (`torch.onnx.symbolic_opset9`) `__all__`
    - To make it NOT look public: make its name start with `_`
```

titaiwangms committed:
Update on "[ONNX] Add aten::zero support"


Fixes #90268 

When we use `tensor.zero_()` with inplace slice, it actually uses `aten::zero` instead. 

[ghstack-poisoned]

titaiwangms committed:
Update on "[ONNX] Add aten::zero support"


Fixes #90268 

When we use `tensor.zero_()` with inplace slice, it actually uses `aten::zero` instead. 

[ghstack-poisoned]

titaiwangms commented:
@pytorchbot merge -g
